db.php

```diff
@@ -170,13 +170,20 @@ private function reEncryptPasswords($userId, $oldMasterPass) {
 
     // This method retrieves the user ID for the currently logged-in user
     public function getUserId() {
-        $query = $this->database->select('users', [
-            'user_id'
-        ], [
-            'username' => $_SESSION['username']
-        ]);
+        if (!isset($_SESSION['userId'])) {
+            $query = $this->database->select('users', [
+                'user_id'
+            ], [
+                'username' => $_SESSION['username']
+            ]);
+
+            $userId = $query[0]['user_id'];
```

Comment:
you should check if the query is not empty.
Otherwise this function will throw an error.

Reply:
At first we should catch the exception and log it.
Afterwards we can still implement an information